samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -0,0 +1,598 @@
+
+            Write-Verbose ($lc | Out-String) 
+
+            function ChangeLab ($lab) {
+                $htvars.Keys | ForEach-Object { $lab.($_) = $htvars[$_]}
+            }
+            $lc | ForEach-Object { ChangeLab  $_}
+            $lc
+        }
+    }
+}
+
+function Show-LabMenu {
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Array containing one line for each lab to be created", ValueFromPipeline = $true)]
+        [ValidateNotNullOrEmpty()]
+        [psobject[]]
+        $labs,
+
+        [Parameter(Mandatory = $false, HelpMessage = "Pick one lab from the labs' list")]
+        [switch]
+        $PickLab,
+
+        [Parameter(Mandatory = $false, HelpMessage = "Which lab properties to show a prompt for")]
+        [ValidateNotNullOrEmpty()]
+        [string[]]
+        $Properties
+    )
+
+    begin {
+
+        function LabToString($lab, $index) {
+            return "[$index]`t$($lab.Id)`t$($lab.ResourceGroupName)`t$($lab.LabName)"
+        }
+
+        $propsPassed = $PSBoundParameters.ContainsKey('Properties')
+        $pickLabPassed = $PSBoundParameters.ContainsKey('PickLab')
+
+        if($pickLabPassed) {
+           Write-Host "LABS"
+        }
+
+        $aggregateLabs = @()
+    }
+    process {
+        $aggregateLabs += $labs
+    }
+    end {
+
+        if($pickLabPassed) {
+            $index = 0
+            $aggregateLabs | ForEach-Object { Write-Host (LabToString $_ ($index++)) }
+
+            $resp = $null
+            do {
+                $resp = Read-Host -Prompt "Please select the lab to create"
+                $resp = $resp -as [int]
+                if($resp -eq $null) {
+                    Write-Host "Not an integer.Try again." -ForegroundColor red
+                }
+                if($resp -and ($resp -ge $labs.Length -or $resp -lt 0)) {
+                    Write-Host "The lab number must be between 0 and $($labs.Length - 1). Try again." -ForegroundColor red
+                    $resp = $null
+                }
+            } until ($resp -ne $null)
+            $aggregateLabs = ,$aggregateLabs[$resp]
+        }
+
+        if($propsPassed) {
+            $hash = @{}
+            $properties | ForEach-Object { $hash[$_] = Read-Host -Prompt "$_"}
+
+            $aggregateLabs = $aggregateLabs | Set-LabProperty @hash
+        }
+        return $aggregateLabs
+    }
+}
+
+# I am forced to use parameter names starting with 'An' because otherwise they get
+# bounded automatically to the fields in the CSV and added to $PSBoundParameters
+function Select-Lab {
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Array containing one line for each lab to be created", ValueFromPipeline = $true)]
+        [ValidateNotNullOrEmpty()]
+        [psobject[]]
+        $labs,
+
+        [parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true, HelpMessage = "Id to look for")]
```

Comment:
Can we say "SearchString" instead of "AnId"?  In this case what do we mean by the "Id"?  The lab won't have an ID in the CSV file, but it has a LabName and other potentially interesting fields that could be used for selecting a set of labs...